### PR TITLE
feat/ovos_plugin_manager

### PR DESCRIPTION
**Pull request by JarbasAl**
removes most hardcoded engines for hotwords/audio/tts/stt in favor of the plugin system

removes code for the XXFactory objects, now imported from ovos plugin manager

might make sense to hold on merging this until all engines have an equivalent plugin mapped in opm factory classes for full backwards compat. but given that most engines would already fail unless requirements were manually installed i dont think this matters. all defaults are accounted for

related issues:
 - https://github.com/MycroftAI/mycroft-core/issues/2874
 - https://github.com/MycroftAI/mycroft-core/issues/2765
 - https://github.com/MycroftAI/mycroft-core/issues/2701

### CODE REVIEW

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

File: requirements/minimal.txt [hunk 1]
```diff
@@ -6,4 +6,5 @@ mycroft-messagebus-client~=0.9.1,!=0.9.2,!=0.9.3
 inflection~=0.3.1
 psutil~=5.6.6
 fasteners~=0.14.1
-requests-futures ~= 0.9.5
\ No newline at end of file
+requests-futures ~= 0.9.5
+ovos-plugin-manager>=0.0.2a2
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply requirements/minimal.txt:10]: 0.0.2 published

File: requirements/extra-audiobackend.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 pychromecast==3.2.2
 python-vlc>=1.1.2
-#git+https://github.com/JarbasAl/py_mplayer
\ No newline at end of file
+# TODO add ovos ocp + ovos vlc
```
Comment (JarbasAl) [inline requirements/extra-audiobackend.txt:3]: this TODO is for a future PR, note that vlc and simple audio backends have not been removed by this PR

**Review by NeonDaniel — APPROVED** [review]

File: .github/workflows/build_tests.yml [hunk 3]
```diff
@@ -19,10 +19,15 @@ jobs:
           python -m pip install build wheel
       - name: Install System Dependencies
         run: |
-          sudo apt install python3-dev swig libssl-dev libfann-dev portaudio19-dev
+          sudo apt-get update
+          sudo apt install python3-dev swig libssl-dev libfann-dev portaudio19-dev libpulse-dev
       - name: Build Distribution Packages
         run: |
           python setup.py bdist_wheel
+      - name: Install tflite_runtime workaround tflit bug
```
Comment (NeonDaniel) [inline .github/workflows/build_tests.yml:27]: Does this deserve an issue?

Comment (JarbasAl) [reply .github/workflows/build_tests.yml:27]: maybe, there is one in the upstream tflit repo already, happens in some platforms due to some mismatch with post release version of the wheel it tries to download or something during setup.py

Comment (NeonDaniel) [reply .github/workflows/build_tests.yml:27]: Just want to make sure we remember to remove this when upstream gets fixed, though it's not a big deal just for automation

**Pull request merged**